### PR TITLE
[15.0][FIX] mail_tracking: failed messages in thread

### PR DESCRIPTION
**Pull request by chienandalu**
- Recover the failed messages box in the mail.thread
- Fix failed message box template issues
- Remove dead code
- Fix send() method signature
- Fix patch methods js files paths
- Update roadmap

Peek 09-08-2023 13-32

cc @Tecnativa TT44683

please review @pedrobaeza @CarlosRoca13

### CODE REVIEW

**Review by pedrobaeza — APPROVED** [review]
Thanks for the fixes and improvements.

**Review by CarlosRoca13 — APPROVED** [review]

File: mail_tracking/models/mail_thread.py [hunk 1]
```diff
@@ -99,46 +99,35 @@ def _fields_view_get(
...
-                nodes[0].append(etree.Element("separator"))
-                nodes[0].append(new_filter)
-        elif view_type == "form":
-            # Modify view to add new field element
-            nodes = doc.xpath("//field[@name='message_ids' and @widget='mail_thread']")
-            if nodes:
-                # Create field
-                field_failed_messages = etree.Element(
-                    "field",
-                    {"name": "failed_message_ids", "widget": "mail_failed_message"},
```
Comment (pedrobaeza) [inline mail_tracking/models/mail_thread.py:140]: Don't you need to have the field in the form for using it?

Comment (chienandalu) [reply mail_tracking/models/mail_thread.py:140]: No, it's directly injected before the attachments box component

**Comment by pedrobaeza** [comment]
/ocabot merge minor

Please supersede the migration to v16 for finishing it with these changes.

**Pull request merged**